src/components/Pickers/Suggestion/SuggestionController.ts

```diff
@@ -0,0 +1,85 @@
+export interface ISuggestionModel {
+  item: any;
+  isSelected: boolean;
+}
+
+export class SuggestionController {
+  public currentIndex: number;
+  public currentSuggestion: ISuggestionModel;
+  private suggestions: ISuggestionModel[];
+  constructor() {
+      this.suggestions = [];
+      this.currentIndex = -1;
+  }
+
+  public updateSuggestions(newSuggestions: any[]) {
+    if (newSuggestions && newSuggestions.length > 0) {
+      this.suggestions = this._convertSuggestionsToSuggestionItems(newSuggestions);
+      this.currentIndex = 0;
+      this.suggestions[0].isSelected = true;
+      this.currentSuggestion = this.suggestions[0];
+    } else {
+      this.suggestions = [];
+      this.currentIndex = -1;
+      this.currentSuggestion = undefined;
+    }
+  }
+
+  public nextSuggestion(): boolean {
+    if (this.currentIndex < (this.suggestions.length - 1)) {
+      this.setSelectedSuggestion(this.currentIndex + 1);
+      return true;
+    } else if (this.currentIndex === (this.suggestions.length - 1)) {
+      this.setSelectedSuggestion(0);
+      return true;
+    }
+  }
```

Comment:
I'm surprised this doesn't error/warn for a missing return. Be explicit in your fallback.